src/tools/check_version.ts

```diff
@@ -0,0 +1,95 @@
+import { execSync } from 'child_process';
+import axios from 'axios';
+
+/**
+ * A function to make an API call to open VSX in order to get details about the latest release of redhat.ansible extension
+ * @param currentVersion string representing the current version against which information is required
+ * @returns version of the latest release
+ */
+function makeGetRequest() {
+  return new Promise(function (resolve, reject) {
+    axios
+      .get('https://open-vsx.org/api/redhat/ansible')
+      .catch(function (error) {
+        if (error.response) {
+          // Request made and server responded
+          console.log(error.response.status);
+          resolve(error.response.status);
+        } else if (error.request) {
+          // The request was made but no response was received
+          console.log(error.request);
+          reject(error.request);
+        } else {
+          // Something happened in setting up the request that triggered an Error
+          console.log('Error', error.message);
+          reject(error.message);
+        }
+      })
+      .then((response) => {
+        if (response) {
+          const result = response.data.version;
+          resolve(result);
+        }
+      });
+  });
+}
+
+/**
+ * A function that makes a call to the makeGetRequest and checks if the curren version of redhat.ansible is released in Open VSX or not
+ * @param currentVersion string representing the current version against which information is required
+ * @returns true if current version is released, false otherwise
+ */
+
+async function releasedInOpenVSX(currentVersion: string): Promise<boolean> {
+  const responseVersion = await makeGetRequest();
+
+  return true ? responseVersion === currentVersion : false;
```

Comment:
This looks rather pointless. What are you doing here exactly?

Reply:
Thanks. I confused it with Python's if-else.